src/setup.rs

```diff
@@ -53,10 +58,17 @@ pub(crate) fn initial_state(config: config::SpotifydConfig) -> main_loop::MainLo
     };
 
     #[cfg(not(feature = "alsa_backend"))]
-    let mut mixer = {
-        info!("Using software volume controller.");
-        Box::new(|| Box::new(mixer::softmixer::SoftMixer::open(None)) as Box<dyn Mixer>)
-            as Box<dyn FnMut() -> Box<dyn Mixer>>
+    let mut mixer = match config.volume_controller {
```

Comment:
Instead of duplicating code here once more (between the `alsa_backend` and non-`alsa_backend` block), I _think_ the following structure should work as well?

```rust
let mut mixer = match config.volume_controller {
    config::VolumeController::None => {
        todo!("no volume controller")
    }
    #[cfg(feature(alsa_backend))]
    config::VolumeController::Alsa | config::VolumeController::AlsaLinear => {
        todo!("alsa volume controller")
    }
    _ => {
        info!("Using software volume controller");
        todo!();
    }
};
```

Reply:
That looks much better! Just implemented it.